cdi/src/main/java/io/smallrye/config/inject/ConfigPropertiesInjectionBean.java

```diff
@@ -52,8 +53,16 @@ public T create(final CreationalContext<T> creationalContext) {
             }
         }
 
-        SmallRyeConfig config = (SmallRyeConfig) ConfigProvider.getConfig(getContextClassLoader());
-        return config.getConfigMapping(getBeanClass(), prefix);
+        Config config = ConfigProvider.getConfig(getContextClassLoader());
+        SmallRyeConfig smallRyeConfig;
+
+        if (config instanceof SmallRyeConfig) {
+            smallRyeConfig = (SmallRyeConfig) config;
+        } else {
+            smallRyeConfig = config.unwrap(SmallRyeConfig.class);
+        }
+
+        return smallRyeConfig.getConfigMapping(getBeanClass(), prefix);
```

Comment:
Couldn't something like this work?

```suggestion
        SmallRyeConfig config = ConfigProvider.getConfig(getContextClassLoader().unwrap(SmallRyeConfig.class);
        return config.getConfigMapping(getBeanClass(), prefix);
```

Reply:
Yep, as I mentioned the instanceof check was just to keep a more similar codepath if it was already a `SmallRyeConfig`, but we should be able to use unwrap directly with no checking.

Reply:
Changed.